src/command/cmd_funcs.c

```diff
@@ -9296,33 +9296,74 @@ cmd_mam(ProfWin* window, const char* const command, gchar** args)
 gboolean
 cmd_editor(ProfWin* window, const char* const command, gchar** args)
 {
-    const char* filename = "/tmp/profanity.txt";
+    /* Build temp file name */
+    GString * tempfile = g_string_new(g_get_tmp_dir());
+    g_string_append(tempfile, "/profanity-");
+    xmpp_ctx_t* const ctx = connection_get_ctx();
+    if(!ctx) {
+        cons_show("Editor: Not connection");
+        return TRUE;
+    }
+    char* uuid = xmpp_uuid_gen(ctx);
+    g_string_append(tempfile, uuid);
+    if (uuid) {
+        xmpp_free(ctx, uuid);
+    }
+    g_string_append(tempfile, ".txt");
+
+    // tempfile should be something like
+    // /tmp/profanity-f2f271dd-98c8-4118-8d47-3bd49c8e2e63.txt
+    const char* filename = tempfile->str;
+
+    // Check if file exists and create file
+    if (g_file_test(filename, G_FILE_TEST_EXISTS)) {
+        cons_show("Editor: The temp file exists");
+        g_string_free (tempfile, TRUE);
+        return TRUE;
+    }
+
+    GError* creation_error = NULL;
+    GFile* file = g_file_new_for_path(filename);
+    GFileOutputStream* fos = g_file_create (file,
+               G_FILE_CREATE_PRIVATE, NULL,
+               &creation_error);
+    if ( creation_error ) {
+        cons_show("Editor: Error during file creation");
+        return TRUE;
+    }
+    g_object_unref(fos);
+
+    // Fork / exec
     pid_t pid = fork();
     if( pid == 0 ) {
-        int x = execl("/usr/bin/vim", "/usr/bin/vim", filename, (char *) NULL);
+        int x = execl("/usr/bin/sensible-editor", "/usr/bin/sensible-editor", g_file_get_path(file), (char *) NULL);
```

Comment:
sensible-terminal is a Debian thing. We probably want to set an executable with `/executables` and default to `$EDITOR`.

Reply:
If you don't have time to do this one just leave it and I'll do it myself after merging.

Reply:
Maybe /usr/bin/vi is a good idea. I would not use $EDITOR right now. I have not define a specification of $EDITOR / $VISUAL Maybe VISUAL would be much better than EDITOR.
I think we can change it later.

Reply:
done in 8be46422d2ca5afff675ef3f7a2117c097a642a7